File: ExampleApp/app/src/main/java/com/motiv/example/dao/PhotosListResponseDao.java
package com.motiv.example.dao;

import androidx.room.*;
import com.motiv.example.PhotosListResponseWithReferences;

@Dao
public interface PhotosListResponseDao {

    @Insert
    public long savePhotosListResponse(com.motiv.example.PhotosListResponse photosListResponse);

    @Delete
    public int deletePhotosListResponse(com.motiv.example.PhotosListResponse photosListResponse);

    @Query("SELECT * FROM photosListResponse")
    @Transaction
    public java.util.List<PhotosListResponseWithReferences> getPhotosListResponses();

    @Query("SELECT * FROM photosListResponse WHERE id = :id")
    @Transaction
    public PhotosListResponseWithReferences getPhotosListResponse(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void savePhotosListResponses(
            java.util.List<com.motiv.example.PhotosListResponse> photosListResponse);
}

File: ExampleApp/app/src/main/java/com/motiv/example/dao/PhotosListResponseDao.kt
 
package com.motiv.example.dao
import androidx.fragment.app.*
import androidx.room.*
import com.motiv.example.PhotosListResponse
import com.motiv.example.PhotosListResponseWithReferences
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*
import kotlin.collections.List
@Dao
public interface PhotosListResponseDao {

    @Insert
    public fun savePhotosListResponse(photosListResponse: com.motiv.example.PhotosListResponse): Long@Delete
    public fun deletePhotosListResponse(photosListResponse: com.motiv.example.PhotosListResponse): Int@Query("SELECT * FROM photosListResponse")
    @Transaction
    public fun getPhotosListResponses(): List<PhotosListResponseWithReferences>@Query("SELECT * FROM photosListResponse WHERE id = :id")
    @Transaction
    public fun getPhotosListResponse(id: String): PhotosListResponseWithReferences@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun savePhotosListResponses(photosListResponse: List<com.motiv.example.PhotosListResponse>): Unit
}
